openmetadata-service/src/main/java/org/openmetadata/service/security/saml/OMMicrometerHttpFilter.java

```diff
@@ -39,16 +41,23 @@ public void init(FilterConfig filterConfig) throws ServletException {
 
   public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
       throws IOException, ServletException {
-    long startTime = System.nanoTime();
-    chain.doFilter(request, response);
-    double elapsed = (double) (System.nanoTime() - startTime) / 1.0E9;
-    String requestPath = ((HttpServletRequest) request).getPathInfo();
-    if (CommonUtil.nullOrEmpty(requestPath)) {
-      requestPath = ((HttpServletRequest) request).getServletPath();
-    }
-    String responseStatus = String.valueOf(((HttpServletResponse) response).getStatus());
-    String requestMethod = ((HttpServletRequest) request).getMethod();
-    MicrometerBundle.httpRequests.labels(requestMethod, responseStatus, requestPath).observe(elapsed);
+    webAnalyticEvents.record(
+        () -> {
+          long startTime = System.nanoTime();
+          try {
+            chain.doFilter(request, response);
+          } catch (IOException | ServletException e) {
```

Comment:
what are we doing here ?? the method already throws IOException and ServletException , why we need to catch it and throw RuntimeException?